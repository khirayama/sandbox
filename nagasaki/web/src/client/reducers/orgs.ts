import { Actions } from '../actions/orgs';

export interface State {
  name: string;
  repos: Array<{
    url: string;
    name: string;
    language: string;
    forksCount: number;
    issuesCount: number;
    stargazersCount: number;
    watchersCount: number;
  }>;
  isFetching: boolean;
}

export const initialState: State = {
  name: '',
  repos: [],
  isFetching: false
};

export const reducer = (state: State = initialState, action: Actions): State => {
  switch (action.type) {
    case 'FETCH_REPOS':
      return { ...state, isFetching: true };
    case 'FETCH_REPOS_SUCCESS':
      return { ...state, isFetching: false, ...action.payload };
    case 'RESET_ORGS':
      return initialState;
    default:
      return state;
  }
};
