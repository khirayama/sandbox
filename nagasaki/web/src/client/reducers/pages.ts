import { Actions } from 'client/actions/pages';

export interface State {
  isLoadingCompletion: boolean;
  error: Error | null;
}

export const initialState: State = {
  isLoadingCompletion: false,
  error: null
};

export const reducer = (state: State = initialState, action: Actions): State => {
  switch (action.type) {
    case 'RESET_PAGES_STATUS':
      return initialState;
    case 'LOAD_TOP_PAGE':
      return { ...state, isLoadingCompletion: false };
    case 'LOAD_TOP_PAGE_SUCCESS':
      return { ...state, isLoadingCompletion: true };
    case 'LOAD_TOP_PAGE_FAILURE':
      return { ...state, isLoadingCompletion: true, error: action.payload.err };
    default:
      return state;
  }
};
