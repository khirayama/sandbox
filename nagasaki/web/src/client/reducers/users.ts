import { Actions } from '../actions/users';

export interface State {
  name: string;
}

export const initialState: State = {
  name: ''
};

export const reducer = (state: State = initialState, action: Actions): State => {
  switch (action.type) {
    case 'SET_USER_NAME':
      return { ...state, name: action.payload.name };
    default:
      return state;
  }
};
