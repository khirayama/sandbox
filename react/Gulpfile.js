var gulp = require('gulp');
var gReact = require('gulp-react');
var gReplace = require('gulp-replace');
var browserify = require('browserify');
var source = require('vinyl-source-stream');
var del = require('del');
var runSequence = require('run-sequence');

var browserifyConfig = {
  entries: ['./index.js'],
  standalone: 'Flux'
};

gulp.task('clean', function(cb) {
  del(['lib/', 'Flux.js'], cb);
});

gulp.task('lib', function() {
  return gulp.src('src/*.js')
             .pipe(gReact({harmony: true}))
             .pipe(gReplace(/__DEV__/g, 'false'))
             .pipe(gulp.dest('lib'));

});

gulp.task('browserify', ['lib'], function() {
  return browserify(browserifyConfig)
          .bundle()
          .pipe(source('Flux.js'))
          .pipe(gulp.dest('./dist/'))
});

gulp.task('build', ['lib', 'browserify']);

gulp.task('publish', function(cb) {
  runSequence('clean', 'build', cb);
});

gulp.task('default', ['build']);
