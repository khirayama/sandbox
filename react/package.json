{
  "name": "flux",
  "version": "2.0.2",
  "description": "An application architecture based on a unidirectional data flow",
  "keywords": [
    "flux",
    "react",
    "facebook",
    "dispatcher"
  ],
  "homepage": "http://facebook.github.io/flux/",
  "bugs": "https://github.com/facebook/flux/issues",
  "files": [
    "index.js",
    "lib/",
    "LICENSE",
    "PATENTS"
  ],
  "main": "index.js",
  "scripts": {
    "prepublish": "gulp publish",
    "test": "jest"
  },
  "jest": {
    "rootDir": "src",
    "scriptPreprocessor": "../jest-preprocessor.js"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/facebook/flux"
  },
  "author": "Facebook",
  "contributors": [
    "Jing Chen <jingc@fb.com>",
    "Bill Fisher <fisherwebdev@gmail.com>",
    "Paul O'Shannessy <paul@oshanessy.com>"
  ],
  "license": "BSD",
  "devDependencies": {
    "browserify": "^9.0.3",
    "del": "^1.1.1",
    "gulp": "^3.8.11",
    "gulp-clean": "^0.3.1",
    "gulp-react": "^2.0.0",
    "gulp-replace": "^0.5.3",
    "jest-cli": "^0.4.0",
    "react-tools": "^0.12.0",
    "run-sequence": "^1.0.2",
    "vinyl-source-stream": "^1.0.0"
  }
}
