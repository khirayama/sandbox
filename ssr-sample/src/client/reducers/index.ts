import { Action } from 'redux';

export interface State {
  count: number;
  ui: {
    locale: 'en' | 'ja';
  };
}

export const initialState: State = {
  count: 1,
  ui: {
    locale: 'en',
  },
};

export function reducer(state = initialState, action: any) {
  switch (action.type) {
    case 'INCREMENT': {
      return {
        count: state.count + 1,
        ui: state.ui,
      };
      break;
    }
    case 'DECREMENT': {
      return {
        count: state.count - 1,
        ui: state.ui,
      };
      break;
    }
    case 'CHANGE_LOCALE': {
      return {
        count: state.count,
        ui: {
          locale: action.payload.locale,
        },
      };
      break;
    }
    default:
  }
  return state;
}
