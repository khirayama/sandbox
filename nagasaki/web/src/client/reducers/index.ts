import { combineReducers } from 'redux';
import { RouterState } from 'connected-react-router';
import { reducer as usersReducer, State as UsersState } from './users';
import { reducer as orgsReducer, State as OrgsState } from './orgs';
import { reducer as pagesReducer, State as PagesState } from './pages';

export interface State {
  users: UsersState;
  orgs: OrgsState;
  pages: PagesState;
  router: RouterState;
}

export const rootReducer = combineReducers({
  users: usersReducer,
  orgs: orgsReducer,
  pages: pagesReducer
});
