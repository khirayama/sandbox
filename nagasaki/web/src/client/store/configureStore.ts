import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import * as connectedReactRouter from 'connected-react-router';
import * as history from 'history';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import { createRootReducer } from '../reducers';
import { rootSaga } from '../sagas';

export const his = process.env.IS_BROWSER ? history.createBrowserHistory() : history.createMemoryHistory();
const sagaMiddleware = createSagaMiddleware();

const createEnhancer = () => {
  const composeEnhancers = composeWithDevTools({});

  return composeEnhancers(applyMiddleware(sagaMiddleware, connectedReactRouter.routerMiddleware(his)));
};

export const runSaga = async () => {
  // return sagaMiddleware.run(rootSaga).done;
  return sagaMiddleware.run(rootSaga).toPromise;
};

export const configureStore = (preloadedState: Object = {}) => {
  const enhancer = createEnhancer();
  const store = createStore(createRootReducer(his), preloadedState, enhancer);

  runSaga();

  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('../reducers', () => {
      const {
        rootReducer: nextReducer
      }: { rootReducer: typeof createRootReducer } = require('../reducers');

      store.replaceReducer(createRootReducer(his));
    });
  }

  return store;
};
