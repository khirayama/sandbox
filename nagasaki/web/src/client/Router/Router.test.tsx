import * as React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import configureStore from 'redux-mock-store';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import { rootReducer } from '../reducers';
import { Router } from '.';

let state: any;
let store: any;

beforeEach(() => {
  state = createStore(rootReducer).getState();
  store = configureStore()(state);
});

test('should render Top page', () => {
  const tree = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
        <Router />
      </MemoryRouter>
    </Provider>
  );

  expect(toJson(tree)).toMatchSnapshot();
});

test('should render Orgs page', () => {
  const tree = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={[{ pathname: '/orgs/test', key: 'testKey' }]}>
        <Router />
      </MemoryRouter>
    </Provider>
  );

  expect(toJson(tree)).toMatchSnapshot();
});

test('should render 404 page', () => {
  const tree = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={[{ pathname: '/404', key: 'testKey' }]}>
        <Router />
      </MemoryRouter>
    </Provider>
  );

  expect(toJson(tree)).toMatchSnapshot();
});
