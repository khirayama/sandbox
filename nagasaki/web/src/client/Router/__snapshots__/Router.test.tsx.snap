// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 404 page 1`] = `
Array [
  <header
    class="sc-bwzfXH jsidzM"
  >
    <a
      class="sc-bdVaJa kuRBwa"
      href="/"
    >
      &lt;
    </a>
    <span />
    <span />
  </header>,
  <div
    class="sc-htpNat gYfjpC"
  />,
]
`;

exports[`should render Orgs page 1`] = `
Array [
  <header
    class="sc-bwzfXH jsidzM"
  >
    <a
      class="sc-bdVaJa kuRBwa"
      href="/"
    >
      &lt;
    </a>
    <span />
    <span />
  </header>,
  <div
    class="sc-htpNat gYfjpC"
  />,
]
`;

exports[`should render Top page 1`] = `
Array [
  <header
    class="sc-bwzfXH jsidzM"
  >
    <a
      class="sc-bdVaJa kuRBwa"
      href="/"
    >
      &lt;
    </a>
    <span />
    <span />
  </header>,
  <div
    class="sc-htpNat gYfjpC"
  />,
]
`;
